### ISSUE TITLE
[v1.4.6] "Null is not a valid constant value for this type" exception when using default(CancellationToken) in interface

### ISSUE THREAD

**Issue by MaxDeg** [canton7/RestEase#101]
With the latest update of RestEase (1.4.6) we get the following error when trying to create a client on a specific interface.

```
Message: System.ArgumentException : Null is not a valid constant value for this type.
```

```csharp
    public partial interface IEchoApi
    {
    	[RestEase.Get("/echo/{message}")]
    	System.Threading.Tasks.Task<string> Echo([RestEase.Path("message")] string message, CancellationToken cancellationToken = default(CancellationToken));

    	[RestEase.Post("/echo")]
    	[RestEase.Header("Content-Type", "application/json")]
    	System.Threading.Tasks.Task<EchoEntity> EchoWithEntity([RestEase.Body] EchoEntity echoEntity, CancellationToken cancellationToken = default(CancellationToken));    
    }
```

If we remove the ```default(CancellationToken)``` part the exception do not occur.

StackTrace:
System.Reflection.Emit.TypeBuilder.SetConstantValue(ModuleBuilder module, Int32 tk, Type destType, Object value)
   at RestEase.Implementation.ImplementationBuilder.HandleMethods(TypeBuilder typeBuilder, Type interfaceType, FieldBuilder requesterField, FieldInfo classHeadersField, AllowAnyStatusCodeAttribute classAllowAnyStatusCodeAttribute, SerializationMethodsAttribute classSerializationMethodsAttribute, PropertyGrouping properties, MethodInfoGrouping& methodInfoGrouping)
   at RestEase.Implementation.ImplementationBuilder.BuildImplementationImpl(Type interfaceType)
   at RestEase.Implementation.ImplementationBuilder.CreateImplementation[T](IRequester requester)

It looks like the error is coming from here:
https://github.com/canton7/RestEase/commit/d511c244ea976f8a918c762635176a3fd818e1c2#diff-8f33c73c750541c6caed52507bb252f9R358

**Comment by canton7** [comment]
Looks like the same issue as here: https://github.com/dotnet/corefx/issues/26184

**Comment by canton7** [comment]
Mind giving this a test? It passes my unit test, but I want to be sure. Thanks! https://ci.appveyor.com/project/canton7/restease/build/1.4.6.319/artifacts

**Comment by MaxDeg** [comment]
Sure :)
I tested on my solution and your fix it's working.

Thanks

**Comment by canton7** [comment]
Released as v1.4.7. Sorry about that!

**Comment by MaxDeg** [comment]
Great :) Thank you
